### DISCUSSION TITLE
Add an option to use the git-credentials file instead of inlining the token in the URL

### DISCUSSION THREAD

**Discussion by azro352** [renovatebot/renovate#25303] [Suggest an Idea]
### Tell us more.

Hi, 

### Problem

My organization gitlab is configured with Kerberos thus basic HTTP authentication doesn’t work (shitty configuration I know, but I can't change that), the following doesn't work

```bash
git clone https://oauth2:glpat-TOKENTOKEN@my.gitlab.myorg/mygroup/myrepo.git
```

### Solution

There is a workaround using `git-credentials` file
```bash
git config --global credential.helper store
echo "https://oauth2:glpat-TOKENTOKEN@my.gitlab.myorg/mygroup/myrepo.git" > ~/.git-credentials
git clone https://my.gitlab.myorg/mygroup/myrepo.git
```

I tried using this in the renovate docker but as renovate adds an auth part, it overrides the use of the git-credential and fails https://github.com/renovatebot/renovate/blob/58cc2c073738daca066ed87130099f59630c6d9f/lib/modules/platform/gitlab/index.ts#L290

### Proposal

Would you agree to add an option "use-git-credentials-file" to avoid setting the auth part in the url ?

```ts
if(!config.use_git_credentials_file) {
  repoUrl.auth = `oauth2:${opts.token!}`;
}
```

I've validated this (by changed the `js` and `js.map` piece of code with ```//poUrl.auth = `oauth2:${opts.token!}`;``` to avoid breaking the map file)

Thanks

**Comment by rarkins** [comment]
We use camelCase, to it should be called `platformGitCredentialsFile` and be passed as part of `PlatformParams` to `initPlatform()`.

**Reply by azro352** [reply]
Is that a yes ? I'll make the PR for that

**Reply by rarkins** [reply]
Yes, PR welcome

**Reply by azro352** [reply]
I don't need it in the `platform/gitlab/index.ts:initPlatform`, but in the `platform/gitlab/index.ts:getRepoUrl:290` 

As the whole config is available to `initRepo`, the caller of `getRepoUrl`, it would be an idea to get it in the method argument, but it is typed with `RepoParams` and it seems a strange interface to declare `platformGitCredentialsFile` there, but it is the place where I need it

```diff
export async function initRepo({
  repository,
  cloneSubmodules,
  ignorePrAuthor,
  gitUrl,
  endpoint,
  includeMirrors,
+   platformGitCredentialsFile,
}: RepoParams): Promise<RepoResult> {
```


It would then need to add a 4th param to `getRepoUrl` being `platformGitCredentialsFile`

```diff
+ const url = getRepoUrl(repository, gitUrl, res, platformGitCredentialsFile);
```

```diff
function getRepoUrl(
  repository: string,
  gitUrl: GitUrlOption | undefined,
  res: HttpResponse<RepoResponse>,
+ platformGitCredentialsFile
): string {
 // ...
  logger.debug(`Using http URL: ${res.body.http_url_to_repo}`);
  const repoUrl = URL.parse(`${res.body.http_url_to_repo}`);
  // TODO: types (#22198)
+ if (!platformGitCredentialsFile) {
      repoUrl.auth = `oauth2:${opts.token!}`;
+ } 
 return URL.format(repoUrl);
}
```

So how do i get the param where I need it ? 


```mermaid
flowchart TB
    start --> 1.globalInitialize
    1.globalInitialize --> initPlatform
    start --> 4.renovateRepository
    4.renovateRepository --> repo.initRepo
    repo.initRepo --> initApis
    initApis --> getPlatformConfig
    getPlatformConfig --> plat.initRepo
    plat.initRepo --> initRepo --> getRepoUrl
```

**Reply by rarkins** [reply]
It's ok to add it to initRepo params
